modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx

```diff
@@ -145,7 +145,10 @@ const SaveFDSFieldsModalContent = ({
 		});
 
 		onSave({
-			createdFDSFields,
+			createdFDSFields: createdFDSFields.map((fdsField) => ({
+				...fdsField,
+				id: Number(fdsField.id),
```

Comment:
How come you had to change this here? Is `id` not a number already?

Reply:
It's not for the JSON returned by our save action. The root cause is that we are [serializing `long` as a string](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/json/JSONObjectImpl.java#L305), and [`id` is a long](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-api/src/main/java/com/liferay/object/model/ObjectEntryModel.java#L117).